src/ast.c

```diff
@@ -326,9 +326,11 @@ static jl_value_t *scm_to_julia_(value_t e, int eo)
 
             e = cdr_(e);
             if (!eo) {
-                if (sym == line_sym && n==1) {
+                if (sym == line_sym && n==2) {
                     return jl_new_struct(jl_linenumbernode_type,
-                                         scm_to_julia_(car_(e),0));
+                                         scm_to_julia_(car_(cdr_(e)),0), // filename
+                                         scm_to_julia_(car_(e),0)        // line
```

Comment:
this line number box is also missing a gc-root during the call to jl_new_struct (looks like it was missing before too :frowning:)